gradio/app.py

```diff
@@ -0,0 +1,314 @@
+    body = await request.json()
+    data = body["data"]
+    action = body["action"]
+    job_hash, queue_position = queueing.push({"data": data}, action)
+    return {"hash": job_hash, "queue_position": queue_position}
+
+
+@app.post("/api/queue/status/", dependencies=[Depends(login_check)])
+async def queue_status(request: Request):
+    body = await request.json()
+    hash = body['hash']
+    status, data = queueing.get_status(hash)
+    return {"status": status, "data": data}
+
+
+########
+# Helper functions
+########
+
+
+def safe_join(directory: str, path: str) -> Optional[str]:
+    """Safely path to a base directory to avoid escaping the base directory.
+    Borrowed from: werkzeug.security.safe_join"""
+    _os_alt_seps: List[str] = list(
+        sep for sep in [os.path.sep, os.path.altsep] if sep is not None and sep != "/"
+    )
+
+    if path != "":
+        filename = posixpath.normpath(path)
+
+    if (
+        any(sep in filename for sep in _os_alt_seps)
+        or os.path.isabs(filename)
+        or filename == ".."
+        or filename.startswith("../")
+    ):
+        return None
+
+    return posixpath.join(directory, filename)    
+
+
+def get_types(cls_set: List[Type], component: str):
+    docset = []
+    types = []
+    if component == "input":
+        for cls in cls_set:
+            doc = inspect.getdoc(cls.preprocess)
+            doc_lines = doc.split("\n")
+            docset.append(doc_lines[1].split(":")[-1])
+            types.append(doc_lines[1].split(")")[0].split("(")[-1])
+    else:
+        for cls in cls_set:
+            doc = inspect.getdoc(cls.postprocess)
+            doc_lines = doc.split("\n")
+            docset.append(doc_lines[-1].split(":")[-1])
+            types.append(doc_lines[-1].split(")")[0].split("(")[-1])
+    return docset, types
+
+
+def get_state():
+    raise DeprecationWarning(
+        "This function is deprecated. To create stateful demos, pass 'state'"
+        "as both an input and output component. Please see the getting started"
+        "guide for more information.")
+
+
+def set_state(*args):
+    raise DeprecationWarning(
+        "This function is deprecated. To create stateful demos, pass 'state'"
+        "as both an input and output component. Please see the getting started"
+        "guide for more information.")
+
+    
+if __name__ == '__main__': # Run directly for debugging: python app.py
+    from gradio import Interface    
+    app.interface = Interface(lambda x: "Hello, " + x, "text", "text",
+                              analytics_enabled=False)
+    app.interface.config = app.interface.get_config_file()
+    app.interface.show_error = True
+    app.interface.flagging_callback.setup(app.interface.flagging_dir)
+    auth = False
+    if auth:
+        app.interface.auth = ("a", "b")
+        app.auth = {"a": "b"}
+        app.interface.auth_message = None
+    else:
+        app.auth = None
+    app.tokens = {}
+    uvicorn.run(app)
```

Comment:
In the debugging mode, I would recommend using several options like reload, and apparent port id, terminal info logs..

Example:
```

uvicorn.run(
            "src.listener:app",
            host="0.0.0.0",
            port=PORT,
            reload=True,
            log_level="info",
        )
```

Reply:
Yup open to that.